modules/rostests/apitests/shell32/ShellExecuteW.cpp

```diff
@@ -0,0 +1,275 @@
+/*
+ * PROJECT:         ReactOS api tests
+ * LICENSE:         GPLv2+ - See COPYING in the top level directory
+ * PURPOSE:         Testing ShellExecuteW
+ * PROGRAMMER:      Doug Lyons <douglyons@douglyons.com>
+ */
+
+#include "shelltest.h"
+#include <stdio.h>
+#include <winbase.h>
+
+/*
+* Returns TRUE on success and FALSE on failure 
+* On success returns the OS Vendor String in VendorString
+*/
+BOOL GetOSVendor(char* VendorString)
+{
+    CHAR file_name[25] = {"OperatingSystem.txt"};
+    CHAR cmdline[100] = {"ver > "};
+    FILE *fp;
+    CHAR OSVendor[100] = {0};
+    CHAR *myreturn;
+    INT pos;
+    char *space_ptr;
+    INT iResult;
+
+    strcpy(VendorString, "Unknown"); // Initialize bad return value
+
+    fp = fopen(file_name, "r"); // read mode
+
+    if(fp != NULL)
+        return FALSE;
+
+    strcat(cmdline, file_name);
+
+    system(cmdline);
+
+    fp = fopen(file_name, "r"); // read mode
+
+    if(fp == NULL)
+        return FALSE;
+
+    /* first we must read past the <CR><LF> at the beginning of the file */
+    myreturn = fgets(OSVendor , 100 , fp);
+
+    if(myreturn == NULL)
+        return FALSE;
+
+    /* Now we can read the first actual text line */
+    myreturn = fgets(OSVendor , 100 , fp);
+
+    if(myreturn == NULL)
+        return FALSE;
+      
+    /* diagnostic only */
+//    printf("OSVendor is '%s'.\n", OSVendor);
+
+    space_ptr = strchr(OSVendor, ' ');
+    pos = -1;
+    if (space_ptr != NULL)
+        {
+            pos = space_ptr - OSVendor;
+        }
+
+    /* diagnostic only */
+//    printf("Space is at position '%d'.\n", pos);
+
+    /* move zero into previous first space character position */
+    if(OSVendor[pos] == 32) OSVendor[pos]=0;
+
+    /* diagnostic only */
+//    printf("The OS Vendor from command is '%s'.\n", OSVendor);
+
+    iResult = fclose(fp);
+
+    if(iResult != 0)
+        return FALSE;
+
+    iResult = remove(file_name);
+
+    if(iResult != 0)
+        return FALSE;
+
+   /* copy our result back to the calling function */
+   strcpy(VendorString, OSVendor);
+
+   return TRUE;
+}
+
+static
+VOID
+TestShellExecuteW()
+{
+    HINSTANCE hInstance;
+    INT retval;
+    CHAR OSVendor[100] = {0};
+    BOOL  IsWindows = FALSE;
+    HWND hWnd;
+    HWND hWnd1;
+    const WCHAR *Name = L"ShellExecuteW";
+
+
+    GetOSVendor(OSVendor);
```

Comment:
Please use `GetWindowsDirectoryW` instead of `GetOSVendor`.

Reply:
Done.